README.md

```diff
@@ -160,6 +160,167 @@ realm.writeBlocking {
 }
 ```
 
+## Observing data changes
+
+Realm support asynchronous observers on all its data structures.
+
+### Realm
+
+A Realm can be observed globally for changes on its data.
+
+```Kotlin
+// Subscribe for change notifications on a Realm instance
+CoroutineScope(context).async {
+    realm.asFlow()
+        .collect { realmChange: RealmChange<Realm> ->
+            when (realmChange) {
+                is InitialRealm<*> -> println("Initial Realm")
+                is UpdatedRealm<*> -> println("Realm updated")
+            }
+        }
+}
+// out: "Initial Realm"
+
+// Write data
+realm.writeBlocking {
+    copyToRealm(Person())
+}
+// out: "Realm updated"
+```
+
+### RealmObject
+
+Realm objects can be observed individually. A list of the changed field names is provided on each update.
+
+```Kotlin
+// Person named Carlo
+val person = realm.writeBlocking {
+    copyToRealm(Person().apply { name = "Carlo" })
+}
+
+// Subscribe for change notifications on person
+CoroutineScope(context).async {
+    person.asFlow().collect { objectChange: ObjectChange<Person> ->
+        when (objectChange) {
+            is InitialObject -> println("Initial object: ${objectChange.obj.name}")
+            is UpdatedObject -> println("Updated object: ${objectChange.obj.name}, changed fields: ${objectChange.changedFields.size}")
+            is DeletedObject -> println("Deleted object")
+        }
+    }
+}
+// out: "Initial object: Carlo"
+
+// Change person field `dog`
+realm.writeBlocking {
+    findLatest(person)?.dog = Dog()
+}
+// out: "Updated object: Carlo, changed fields: 1"
+
+// Delete person
+realm.writeBlocking {
+    findLatest(person)?.delete()
+}
+// out: "Deleted object"
+```
+
+### RealmLists
+
+Realm data structures can be observed too. On `RealmList` on each update you receive what positions were inserted, changed or deleted.
+
+```Kotlin
+// Person named Carlo
+val person = realm.writeBlocking {
+    copyToRealm(Person().apply { name = "Carlo" })
+}
+
+// Subscribe for a RealmList change notifications
+CoroutineScope(context).async {
```

Comment:
Maybe nitpicking a bit here....I really like this part of the docs, but it seems a bit overkill to go through all the states in the README...We should defer to the website docs for that IMO...

It isn't that I'm against having this here, but we should be mindful about the length of the README and this is adding quite a bit to that, and I'm not sure the tradeoff is worth it when we can just show this snippet and otherwise point to the docs?

Reply:
I wasn't sure about how much detail to put in then though on putting the whole test cases. Yeah, it is quite a bit too much for the README. I am going to put in the Readme a snippet of the ReadmeTests.